edb/pgsql/metaschema.py

```diff
@@ -2633,29 +2633,45 @@ class LocalTimeInFunction(dbops.Function):
     """Cast text into time using ISO8601 spec."""
     text = r'''
         SELECT
-            CASE WHEN val !~ (
-                    '^\s*(' ||
-                        '(\d{2}(:\d{2}(:\d{2}(\.\d+)?)?)?|\d{2,6}(\.\d+)?)' ||
-                    ')\s*$'
-                )
+            CASE WHEN date_part('hour', x.t) = 24
             THEN
                 edgedb.raise(
                     NULL::time,
                     'invalid_datetime_format',
                     msg => (
-                        'invalid input syntax for type time: '
+                        'cal::local_time field value out of range: '
                         || quote_literal(val)
-                    ),
-                    detail => (
-                        '{"hint":"Please use ISO8601 format. Examples: '
-                        || '18:43:27 or 18:43 Alternatively '
-                        || '\"to_local_time\" function provides custom '
-                        || 'formatting options."}'
                     )
                 )
             ELSE
-                val::time
-            END;
+                x.t
+            END
+        FROM (
+            SELECT
+                CASE WHEN val !~ (
+                        '^\s*(' ||
+                            '(\d{2}(:\d{2}(:\d{2}(\.\d+)?)?)?|\d{2,6}(\.\d+)?)' ||
+                        ')\s*$'
```

Comment:
Is there any other time format that postgres supports?

Reply:
Postgres is a bit sloppy about digits in between `:`. So it will accept omitted leading 0, like `'2:3:4'::time` is the same as `'02:03:04'::time`, whereas the ISO standard explicitly indicates that "hh:mm:ss" should have leading 0.

Another thing is that we don't accept reduced precision time that omits seconds altogether, while Postgres can accept "hh:mm" format. I forget what the problems were with reduced precision format at the moment.